src/main/scala/com/johnsnowlabs/nlp/util/io/ResourceHelper.scala

```diff
@@ -50,6 +53,38 @@ object ResourceHelper {
         .config("spark.kryoserializer.buffer.max", "1000m")
         .getOrCreate())
 
+  def getSparkSessionWithS3(
+      awsAccessKeyId: String,
+      awsSecretAccessKey: String,
+      awsSessionToken: String): SparkSession = {
+
+    require(
+      SparkSession.getActiveSession.isEmpty,
+      "Spark session already running, can't apply new configuration for S3.")
+
+    SparkSession
+      .builder()
+      .appName("SparkNLP Session with S3 Support")
+      .master("local[*]")
+      .config("spark.driver.memory", "22G")
+      .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+      .config("spark.kryoserializer.buffer.max", "2000M")
+      .config("spark.driver.maxResultSize", "0")
+      .config("spark.jsl.settings.aws.credentials.access_key_id", awsAccessKeyId)
+      .config("spark.jsl.settings.aws.credentials.secret_access_key", awsSecretAccessKey)
+      .config("spark.jsl.settings.aws.credentials.session_token", awsSessionToken)
+      .config("spark.jsl.settings.aws.region", "us-east-1")
+      .config(
+        "spark.hadoop.fs.s3a.aws.credentials.provider",
+        "org.apache.hadoop.fs.s3a.TemporaryAWSCredentialsProvider")
+      .config("spark.hadoop.fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+      .config(
+        "spark.jars.packages",
+        "org.apache.hadoop:hadoop-aws:3.3.1,com.amazonaws:aws-java-sdk:1.11.901")
+      .config("spark.hadoop.fs.s3a.path.style.access", "true")
```

Comment:
Hardcoding some configurations could be risky, in particular the following configurations will change depending on AWS Authentication and S3 configuration of each user:
1. `spark.hadoop.fs.s3a.aws.credentials.provider`: This will depend on the authentication configuration, some users could use MFA (TemporaryAWSCredentialsProvider) others don't
2. `spark.jsl.settings.aws.credentials.session_token`: This is used only for MFA
3. `spark.jsl.settings.aws.region`: Not all users will have S3 on `us-east-1`
4. `spark.jars.packages`: Users may require additional packages. In addition, users have different Hadoop versions, the packages numbers of `org.apache.hadoop:hadoop-aws:3.3.1,com.amazonaws:aws-java-sdk:1.11.901` will change depending on the Hadoop version.
5.  `spark.hadoop.fs.s3a.path.style.acces`: This configuration is only used when S3 directories have dots on it like `auxdata.johnsnowlabs.com` but if the user has a name without dots may not work.